sqlstore/src/test/java/uk/ac/bbsrc/tgac/miso/sqlstore/SQLSecurityProfileDAOTest.java

```diff
@@ -0,0 +1,214 @@
+
+  @Autowired
+  @Spy
+  private JdbcTemplate jdbcTemplate;
+
+  private final SecurityManager securityManager = new SecurityManager() {
+
+    @Override
+    public long saveUser(User arg0) throws IOException {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public long saveGroup(Group arg0) throws IOException {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Collection<User> listUsersByIds(Collection<Long> ids) throws IOException {
+      List<User> users = new ArrayList<>();
+      for (long id : ids) {
+        users.add(getUserById(id));
+      }
+
+      return users;
+    }
+
+    @Override
+    public Collection<User> listUsersByGroupName(String arg0) throws IOException {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Collection<Group> listGroupsByIds(Collection<Long> ids) throws IOException {
+      List<Group> groups = new ArrayList<>();
+      for (long id : ids) {
+        if (id == 2) {
+          groups.add(GROUP);
+        } else {
+          throw new UnsupportedOperationException();
+        }
+      }
+      return groups;
+    }
+
+    @Override
+    public Collection<User> listAllUsers() throws IOException {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Collection<Group> listAllGroups() throws IOException {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public User getUserByLoginName(String arg0) throws IOException {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public User getUserById(Long id) throws IOException {
+      if (id == null) {
+        return null;
+      }
+      User u = new UserImpl();
+      u.setUserId(id);
+      return u;
+    }
+
+    @Override
+    public User getUserByEmail(String arg0) throws IOException {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public SecurityProfile getSecurityProfileById(Long arg0) throws IOException {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Group getGroupByName(String arg0) throws IOException {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Group getGroupById(Long arg0) throws IOException {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Collection<String> getDefaultRoles() {
+      throw new UnsupportedOperationException();
+    }
+  };
+
+  @InjectMocks
+  private SQLSecurityProfileDAO dao;
+
+  @Before
+  public void setUp() throws IOException {
+    MockitoAnnotations.initMocks(this);
+    dao.setJdbcTemplate(jdbcTemplate);
+    dao.setSecurityManager(securityManager);
+  }
+
+  @Test
+  public void testCount() throws IOException {
+    assertTrue("No security profiles in DB", dao.count() > 0);
+  }
+
+  @Test
+  public void testGetByIdNone() throws IOException {
+    assertNull(dao.get(100L).getProfileId());
```

Comment:
Weird that the DAO's get method creates a security profile if it doesn't exist despite the method having an id parameter. I wonder if that is required by something... (observation, not a change request)